### DISCUSSION TITLE
How should I handle "cloning" bodies into another simulation?

### DISCUSSION THREAD

**Discussion by saulo308** [jrouwe/JoltPhysics#786] [Q&A] [answered]
Hello! So, I'm pretty new to JoltPhysics and physics simulations in general.

Basically, I have two simulations running. One simulation has a bouncing sphere on static floor and the other simulation has just a static floor at the same position as the floor on the first simulation. At some point, I want to "clone" this bouncing sphere onto the second simulation. As JoltPhysics can be a deterministic physics engine, how could I create the body on the second simulation as a exact clone for that point forward (i.e., for the next steps both spheres should match 100%, just like a replica)?

Giving a thought to it, I know I need to keep the first sphere state somehow and "restore" it upon creation of the second sphere (on the other simulation). However, I can't seem to find a way to do it, since I need to send the first sphere's state as a serialized string for the second simulation through Socket communication. 

I already have an implementation of getting the first sphere location and creating the second one on the same location (on the second simulation). However, the next step following the creation, they have a location mismatch - showing me that altough I created it on the same location, the body's state differs from one to another.

Any ideas, please?
Thank you!

**Comment by jrouwe** [comment, marked as answer]
Have you read the [Deterministic Simulation](https://jrouwe.github.io/JoltPhysics/) and Rolling Back a Simulation documentation sections?

**Reply by saulo308** [reply]
Wow, I totally missed the "Rolling Back a Simulation" section. I have a starting point now. 

Thank you!
